### ISSUE TITLE
bug: python pypi package can't fetch version

### ISSUE THREAD

**Issue by Trim21** [renovatebot/renovate#6142]
**What Renovate type are you using?**

GitHub App

**Describe the bug**

renovate pr https://github.com/Trim21/pol/pull/181 was autoclosed by renovete this morning, so I check the log and found that all pypi depencency has no version.

**Did you see anything helpful in debug logs?**

log: https://app.renovatebot.com/dashboard#github/Trim21/pol/173326645
```
DEBUG: No pypi result - returning null
{
  "lookupName": "gunicorn",
  "registryUrls": []
}
DEBUG: Failed to look up dependency gunicorn (gunicorn)(packageFile="pyproject.toml", dependency="gunicorn")
DEBUG: pypi dependency not found: fastapi(searching in https://pypi.org/pypi/)
DEBUG: No pypi result - returning null
{
  "lookupName": "fastapi",
  "registryUrls": []
}
DEBUG: Failed to look up dependency fastapi (fastapi)(packageFile="pyproject.toml", dependency="fastapi")
DEBUG: pypi dependency not found: loguru(searching in https://pypi.org/pypi/)
DEBUG: No pypi result - returning null
{
  "lookupName": "loguru",
  "registryUrls": []
}
DEBUG: Failed to look up dependency loguru (loguru)(packageFile="pyproject.toml", dependency="loguru")
DEBUG: pypi dependency not found: python-dateutil(searching in https://pypi.org/pypi/)
DEBUG: No pypi result - returning null
{
  "lookupName": "python-dateutil",
  "registryUrls": []
}
DEBUG: Failed to look up dependency python-dateutil (python-dateutil)(packageFile="pyproject.toml", dependency="python-dateutil")
DEBUG: pypi dependency not found: pydantic(searching in https://pypi.org/pypi/)
DEBUG: No pypi result - returning null
{
  "lookupName": "pydantic",
  "registryUrls": []
}
```

**Comment by viceice** [comment]
[pypi](https://status.python.org/) had some cdn errors yesterday, maybe that was the cause.

Can you check if renovate has run later and again did not found the deps?

**Comment by Trim21** [comment]
@viceice 43 minutes ago No pypi result - returning null

**Comment by viceice** [comment]
can you please paste the logs

**Comment by Trim21** [comment]
https://gist.github.com/Trim21/83aea62d9da5b0644cb7eb9d352c8a1e






------------------&nbsp;Original&nbsp;------------------

**Comment by Trim21** [comment]
https://gist.github.com/Trim21/83aea62d9da5b0644cb7eb9d352c8a1e




------------------&nbsp;Original&nbsp;------------------

**Comment by Trim21** [comment]
https://gist.github.com/Trim21/83aea62d9da5b0644cb7eb9d352c8a1e




------------------&nbsp;Original&nbsp;------------------

**Comment by rarkins** [comment]
Thanks, I've reproduced this using a fork of your repo. I've found the root cause, although not sure how we should fix.

Here's what happens:
- When extracting the poetry dependencies, we note a python constraint of `3.7.*`
- When checking the releases, we see constraints like `requires_python: ">=3.6"`
- We compare those two, and get a *negative* result, so filter out the dep

![image](https://user-images.githubusercontent.com/6311784/81139208-3ad92680-8f65-11ea-944a-01938982e2f3.png)

This feature is designed so that e.g. if a release says `>=3.8` then we would filter it out, because you are using `3.7.*`. However in this case it seems our `matches` algorithm isn't able to handle these two types of ranges.
